### PR TITLE
TST: change pyarrow skips to xfails

### LINKED ISSUE #49411: BUG: "TODO(1.4): Change me into xfail at release time"
There's a bunch of these, some with slightly different wording.  I think they are all greppable with `TODO(1.4)`.  It isn't clear if these were missed at 1.4-release-time or if they were changed and then un-changed post release.  Either way, this is not a good pattern.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #49411

### CODE REVIEW

File: pandas/tests/io/parser/usecols/test_usecols_basic.py [hunk 1]
```diff
@@ -25,8 +25,11 @@
     "Usecols do not match columns, columns expected but not found: {0}"
 )
 
-# TODO: Switch to xfails
-skip_pyarrow = pytest.mark.usefixtures("pyarrow_skip")
+xfail_pyarrow = pytest.mark.usefixtures("pyarrow_xfail")
+
+pytestmark = pytest.mark.filterwarnings(
```
Comment (phofl) [inline pandas/tests/io/parser/usecols/test_usecols_basic.py:30]: Is this used?

Comment (jbrockmendel) [reply pandas/tests/io/parser/usecols/test_usecols_basic.py:30]: yes

**Review by phofl — APPROVED** [review]
lgtm

feel free to merge when ci is finished

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**